react/src/components/atoms/forms/InputNumber/index.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Input from '../Input';
+import { InputContext } from '../Input/context';
+import { validNumber } from '../Input/validate';
+import './style.css';
+
+const Number = (props) => (
+  <React.Fragment>
+    <InputContext.Consumer>
+      {
+        (context) => {
+          const inputClasses = classNames({
+            'ma__input-currency__control': true,
+            'js-is-required': props.required
+          });
+          let errorMsg = '';
+
+          const handleChange = (e) => {
+            const { value } = e.target;
+            const update = { value };
+
+            if (props.required) {
+              errorMsg = 'Please enter a value.';
+              update.showError = true;
+              update.errorMsg = errorMsg;
+            } else if (validNumber(value, props.min, props.max, errorMsg)) {
+              update.showError = true;
+              update.errorMsg = errorMsg;
+            } else {
+              errorMsg = '';
+              update.showError = false;
+              update.errorMsg = errorMsg;
+            }
+
+            context.updateState(update);
+            if (typeof props.onChange === 'function') {
+              props.onChange(e);
+            }
+          };
+
+          const handleAdjust = (e, direction) => {
+            let newValue;
+            if (direction === 'up') {
+              newValue = +context.value + props.step;
```

Comment:
Why is there a plus here?

Reply:
this is to make sure the variable is numeric

Reply:
otherwise it renders a string